modules/utxo-lib/src/transaction.js

```diff
@@ -581,22 +583,48 @@ Transaction.prototype.hashForSignatureByNetwork = function (
   isSegwit,
 ) {
   switch (coins.getMainnet(this.network)) {
-    case networks.bitcoingold:
-      return this.hashForGoldSignature(inIndex, prevoutScript, value, hashType, isSegwit)
-    case networks.bitcoincash:
-    case networks.bitcoinsv:
-      return this.hashForCashSignature(inIndex, prevoutScript, value, hashType)
     case networks.zcash:
       return this.hashForZcashSignature(inIndex, prevoutScript, value, hashType)
-    default:
-      if (isSegwit) {
-        return this.hashForWitnessV0(inIndex, prevoutScript, value, hashType)
-      } else {
-        return this.hashForSignature(inIndex, prevoutScript, hashType)
+    case networks.bitcoingold:
+    case networks.bitcoinsv:
+    case networks.bitcoincash:
+      if (hashType & Transaction.SIGHASH_FORKID) {
```

Comment:
This block of code is some what difficult to follow, as is often the case when using bitwise operators. It would be nice to have more comments here. It looks this PR is removing some comment blocks like what was in `hashForCashSignature` which was helpful for me to understand this PR, so even just restoring those or re-summarizing the intended logic would be good I think. Maybe also putting a description for the `hashType` param.

Reply:
improved comments